com.sun.c1x/src/com/sun/c1x/C1XCompiler.java

```diff
@@ -114,6 +115,9 @@ public CiResult compileMethod(RiResolvedMethod method, int osrBCI, CiStatistics
                 TTY.println(String.format("C1X %4d %-70s %-45s %-50s | %3d.%dms %5dB", index, "", "", "", time / 10, time % 10, result.targetMethod().targetCodeSize()));
             }
         }
+        if (target.arch.isRISCV64() && result.targetMethod().targetCodeSize() >= RISCV64.MAX_DIRECT_JUMP_SIZE) {
+            throw new CiBailout("methods larger than 1MB are not supported in RISCV64: ");
```

Comment:
```suggestion
            throw new CiBailout("methods larger than 1MB are not supported in RISCV64: "  + result.targetMethod());
```
 
Does this work? :point_up: